pennylane/qnodes/passthru.py

```diff
@@ -0,0 +1,145 @@
+    autodiff (AD) framework (such as TensorFlow or PyTorch). This means that the QNode
+    converts all its inputs (which may come in data types specific to the
+    AD framework used, which we denote *ADT* here) into plain Python/NumPy types,
+    computes the required :ref:`quantum function <intro_vcirc_qfunc>` value or Jacobian,
+    and converts the result back into the ADT if necessary.
+
+    In contrast, PassthruQNode works as a white box: it preserves the ADT
+    throughout the computation. This requires that the quantum function is computed
+    using a simulator device that is compatible with the AD framework used (typically
+    implemented using that same framework).
+
+    The advantages of this approach is that the qfunc can be differentiated using its AD framework
+    without requiring a separate method for computing the Jacobian, and that the internals
+    of the simulation are visible in the computational graph.
+
+    PassthruQNode requires some changes to the way other PennyLane components work:
+
+    1. :class:`Operator` must not do domain checking for its parameters, or it must let the ADT pass the check.
+    2. The simulator device must return the result as the ADT instead of plain Python/NumPy types.
+    3. Any output_conversion in :meth:`BaseQNode.evaluate` must be skipped.
+
+    Additionally, any array-like ADT needs to be able to handle (1) scalar multiplication,
+    (2) indexing/slicing, and possibly (3) iteration, as these are the things qfuncs expect of
+    array-like parameters.
+    """
+    def __init__(self, func, device, properties=None):
+        super().__init__(func, device, mutable=True, properties=properties)
+
+    def __repr__(self):
+        """String representation."""
+        detail = "<PassthruQNode: device='{}', func={}, wires={}>"
+        return detail.format(self.device.short_name, self.func.__name__, self.num_wires)
+
+    def _set_variables(self, args, kwargs):
+        # do nothing, since we do not use VariableRefs
+        pass
+
+    def _construct(self, args, kwargs):
+        """Construct the quantum circuit graph by calling the quantum function.
+
+        Like :class:`.BaseQNode._construct`, but does not use VariableRefs.
+
+        PassthruQNode does not have a Jacobian method, so it does not HAVE to use VariableRefs or scalar linear indexing of input parameters.
+        Two options:
+        1. Use VariableRefs anyway, re-use most BaseQNode methods.
+           Problem: after evaluating the VariableRefs, stacking sliced/indexed Tensors in Operation.parameters should somehow result in a Tensor, not an object array.
+        2. Do not use VariableRefs, call the qfunc each time :meth:`PassthruQNode.evaluate` is called (always mutable).
+           Problem: tensornet_tf requires variable_deps?
+
+
+        TODO rethink output_conversion? should require device to return things in a fixed form, but either as arrays or as AD Tensors, do conversion in interface (if necessary...)
```

Comment:
Obviously this docstring is temporary for a WIP implementation. Please remember to update when decisions are finalized